File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/StaffAvailability.swift
import Foundation

// MARK: - Staff Availability Assignment

/// Una asignacion dentro del reporte de disponibilidad.
/// `eventDate` llega como YYYY-MM-DD (sin hora); los turnos son opcionales y
/// llegan en ISO8601 UTC cuando se capturaron.
public struct StaffAvailabilityAssignment: Codable, Sendable, Hashable {
    public let eventId: String
    public let eventName: String
    public let eventDate: String
    public let shiftStart: String?
    public let shiftEnd: String?
    public let status: String

    public init(
        eventId: String,
        eventName: String,
        eventDate: String,
        shiftStart: String? = nil,
        shiftEnd: String? = nil,
        status: String
    ) {
        self.eventId = eventId
        self.eventName = eventName
        self.eventDate = eventDate
        self.shiftStart = shiftStart
        self.shiftEnd = shiftEnd
        self.status = status
    }
}

// MARK: - Staff Availability

/// Disponibilidad agregada por staff. Solo incluye colaboradores con
/// asignaciones en la ventana consultada; lista vacia = todos libres.
public struct StaffAvailability: Codable, Identifiable, Sendable, Hashable {
    public let staffId: String
    public let staffName: String
    public let assignments: [StaffAvailabilityAssignment]

    public var id: String { staffId }

    public init(
        staffId: String,
        staffName: String,
        assignments: [StaffAvailabilityAssignment]
    ) {
        self.staffId = staffId
        self.staffName = staffName
        self.assignments = assignments
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/StaffAvailability.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Respuesta de `GET /api/staff/availability?date=YYYY-MM-DD` o
 * `?start=YYYY-MM-DD&end=YYYY-MM-DD`. Sólo se devuelven colaboradores CON
 * asignaciones en la ventana consultada — si un staff no aparece, está libre.
 */
@Serializable
data class StaffAvailability(
    @SerialName("staff_id") val staffId: String,
    @SerialName("staff_name") val staffName: String,
    val assignments: List<StaffAvailabilityAssignment> = emptyList()
)

@Serializable
data class StaffAvailabilityAssignment(
    @SerialName("event_id") val eventId: String,
    @SerialName("event_name") val eventName: String,
    @SerialName("event_date") val eventDate: String,
    @SerialName("shift_start") val shiftStart: String? = null,
    @SerialName("shift_end") val shiftEnd: String? = null,
    val status: String
)
